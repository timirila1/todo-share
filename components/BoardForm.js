import React, { Component } from 'react';
import { View, StyleSheet, Text } from 'react-native';
import PropTypes from 'prop-types';

import { addBoard, updateBoardById } from '../services/Boards';
import { colors, routes } from '../services/consts';
import { LabeledInput } from './LabeledInput';
import { CustomButton } from './CustomButton';

export class BoardForm extends Component {
    constructor(props) {
        super(props);
        const { board = {} } = this.props;
        this.state = {
            title: board.title,
            desc: board.desc,
            invalidTitle: false,
        };
    }

    isUpdateMode = () => !!(this.props.board && this.props.board.id)

    checkBoardValidity = (data) => {
        if (!data.title) {
            this.setState({ invalidTitle: true });
            return false;
        }
        return true;
    }

    submitForm = async () => {
        const { navigation, board = null } = this.props;
        const data = { ...this.state };

        if (!this.checkBoardValidity(data)) {
            return;
        }

        try {
            if (this.isUpdateMode()) {
                await updateBoardById(board.id, data);
            } else {
                await addBoard(data);
            }
            navigation.navigate(routes.Boards, { refresh: true });
        } catch (e) {
            console.error(e);
        }
    }

    render() {
        const { invalidTitle } = this.state;

        return (
            <View style={styles.container}>
                <View style={styles.fieldset}>
                    <LabeledInput
                        label='Title'
                        value={this.state.title}
                        onChangeText={title => this.setState({ title, invalidTitle: false })}
                        maxLength={40}
                    />
                    {invalidTitle && <Text style={styles.errorMessage}>Please enter Title</Text>}
                </View>
                <View style={styles.fieldset}>
                    <LabeledInput
                        label='Description'
                        multiline={true}
                        numberOfLines={4}
                        value={this.state.desc}
                        onChangeText={desc => this.setState({ desc })}
                    />
                </View>
                <View style={styles.fieldset}>
                    <CustomButton
                        title={this.isUpdateMode()
                            ? 'Update Board'
                            : 'Add Board'}
                        onPress={this.submitForm}
                    />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20,
        paddingLeft: 40,
    },
    fieldset: {
        marginBottom: 20,
    },
    errorMessage: {
        paddingTop: 5,
        color: colors.danger,
    },
});

BoardForm.propTypes = {
    board: PropTypes.object,
    navigation: PropTypes.object,
};
