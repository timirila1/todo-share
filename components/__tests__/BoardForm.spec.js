import 'react-native';
import React from 'react';
import { BoardForm } from '../BoardForm';
import renderer from 'react-test-renderer';

it('Render BoardForm', async () => {
    const tree = renderer.create(
        <BoardForm />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
