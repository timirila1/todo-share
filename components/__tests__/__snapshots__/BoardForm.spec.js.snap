// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BoardForm 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
      "paddingLeft": 40,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.6)",
            },
            false,
          ]
        }
      >
        Title
      </Text>
      <TextInput
        allowFontScaling={true}
        label="Title"
        maxLength={40}
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "borderBottomColor": "#5e1c73",
              "borderBottomWidth": 1,
              "fontSize": 24,
              "paddingBottom": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            false,
            false,
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.6)",
            },
            false,
          ]
        }
      >
        Description
      </Text>
      <TextInput
        allowFontScaling={true}
        label="Description"
        multiline={true}
        numberOfLines={4}
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "borderBottomColor": "#5e1c73",
              "borderBottomWidth": 1,
              "fontSize": 24,
              "paddingBottom": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            false,
            false,
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fa6c35",
          "borderRadius": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
              "fontWeight": "300",
              "lineHeight": 36,
              "textAlign": "center",
            },
            false,
            false,
          ]
        }
      >
        Add Board
      </Text>
    </View>
  </View>
</View>
`;
